### PR TITLE
[Doc] Add a document for tracking Operator/Function support state

### PR DESCRIPTION
**Pull request by zuochunwei**
## What changes were proposed in this pull request?

add a document to record the `Operator/Function` support situation.

This document should be updated when we make any new progress! 

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zuochunwei** [comment]
@rui-mo

**Comment by PHILO-HE** [comment]
Thanks for your work! Just a tiny issue: please capitalize the initial letter for some sentences. Thanks!

File: docs/SupportProgress.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Generally speaking, there are lots of work to do, but all 22 TPC-H queries can be offloaded to the native backend. the information
+is presented separately according to the type of backend. the detail information is as follows:
+
+## Backend-Velox
+
+### Operator support
+Support List
+- TableScan
+- Values
```
Comment (rui-mo) [inline docs/SupportProgress.md:15]: This is not supported in Gluten I think.

File: docs/SupportProgress.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+- Project
+- Filter
+- Aggregate: HashAggregate or StreamingAggregate
```
Comment (rui-mo) [inline docs/SupportProgress.md:18]: StreamingAggregate is also not supported.

File: docs/SupportProgress.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- GroupId
+- HashJoin: HashProbe and HashBuild
+- OrderBy
+- Limit/Offset
+- TopN
```
Comment (rui-mo) [inline docs/SupportProgress.md:23]: I think this is not supported, am I rigth? @JkSelf

File: docs/SupportProgress.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+- Window
+
+Unsupported List
+- Unnest
+- MergeJoin
```
Comment (rui-mo) [inline docs/SupportProgress.md:28]: Supported.

File: docs/SupportProgress.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+- CrossJoin
+- MergeJoin
```
Comment (rui-mo) [inline docs/SupportProgress.md:30]: Duplicated with above.

**Review by rui-mo — COMMENTED** [review]
Thanks for your contribution!

**Comment by zuochunwei** [comment]
> Thanks for your work! Just a tiny issue: please capitalize the initial letter for some sentences. Thanks!

OK, thanks

**Comment by zuochunwei** [comment]
I updated just now, recheck it please @rui-mo

File: docs/SupportProgress.md (hunk 3 above)
Comment (JkSelf) [reply docs/SupportProgress.md]: Yes. TopN is not supported.

File: docs/SupportProgress.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# The Operators and Functions Support Progress
+Although the Gluten project is still in active development, we still have made big progress on Operators and Functions support.
+- The number of Operators supported is 12, the most common Operators are covered.
+- The number of Functions supported is 94, and Velox supports 127 Functions.
```
Comment (rui-mo) [inline docs/SupportProgress.md:4]: @lviiii @zhixingheyi-tian Would you like to help check the function part? Thanks.

File: docs/SupportProgress.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+- The number of Functions commonly used by Spark is 240, and the total number for Spark3.3 is 387.
```
Comment (PHILO-HE) [inline docs/SupportProgress.md:5]: How do we measure commonly used function? Is there a common view in the community or the industry? If not, I think we can just record the overall number here. Thanks!

Comment (FelixYBW) [reply docs/SupportProgress.md:5]: @lviiii mentioned there is a page from Spark documents. @lviiii  where is the page?

File: docs/SupportProgress.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+- OrderBy
+- Limit/Offset
+- MergeJoin
+- Unnest
+- Window
+
+Unsupported List
+- Values
+- TableWrite
+- HashJoin: HashProbe and HashBuild
```
Comment (rui-mo) [inline docs/SupportProgress.md:28]: `HashJoin` is already supported in Gluten, thanks.

Comment (zuochunwei) [reply docs/SupportProgress.md:28]: ok

File: docs/SupportProgress.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+- TableScan
+- Project
+- Filter
+- Aggregate: HashAggregate
+- HashJoin: HashProbe and HashBuild
+- GroupId
+- OrderBy
+- Limit/Offset
+- MergeJoin
+- Unnest
```
Comment (JkSelf) [inline docs/SupportProgress.md:23]: Unnest is not supported.

File: docs/SupportProgress.md (hunk 7 above)
Comment (lviiii) [reply docs/SupportProgress.md:5]: The commonly used functions are referred from [frequently-used built-in functions](https://spark.apache.org/docs/latest/sql-ref-functions.html) , and I add some functions based on my own experience. Maybe, we don't need to mention it. 
The total supported functions' number [for Spark3.3 is 387](https://spark.apache.org/docs/latest/api/sql/) and [for Velox is 204](https://facebookincubator.github.io/velox/functions/coverage.html). Gluten supported 94, in which offloaded 62 is implemented in velox/spark and 32 in velox/presto, shown as below picture.
image

File: docs/SupportProgress.md (hunk 9 above)
Comment (zuochunwei) [reply docs/SupportProgress.md:23]: > Unnest is not supported.

got!

File: docs/SupportProgress.md (hunk 7 above)
Comment (zuochunwei) [reply docs/SupportProgress.md]: updated

**Review by rui-mo — APPROVED** [review]

**Review by PHILO-HE — APPROVED** [review]

**Comment by FelixYBW** [comment]
We need to add the datatype coverage for each function and operator as well. 
@lviiii can you add the detailed table of the function list, datatype and status here?

**Pull request merged**